### ISSUE TITLE
Warning log condition is reversed

### ISSUE THREAD

**Issue by gmtstephane** [deepmap/oapi-codegen#1153]
```golang
e.Use(middleware.OapiRequestValidatorWithOptions(specs, &middleware.Options{
    SilenceServersWarning: true,
}))
```

Output : 

> WARN: OapiRequestValidatorWithOptions called with an OpenAPI spec that has `Servers` set. This may lead to an HTTP 400 with `no matching operation was found` when sending a valid request, as the validator performs `Host` header validation. If you're expecting `Host` header validation, you can silence this warning by setting `Options.SilenceServersWarning = true`. See https://github.com/deepmap/oapi-codegen/issues/882 for more information.

But setting it to false hide the error 

```golang
e.Use(middleware.OapiRequestValidatorWithOptions(specs, &middleware.Options{
    SilenceServersWarning: false,
}))
```

I think the condition to check the silence warning is reversed

**Comment by gmtstephane** [comment]
Addressed in pr #1152

**Comment by jamietanna** [comment]
Thanks for catching, and fixing, this!

**Comment by jamietanna** [comment]
Closed by #1153.
